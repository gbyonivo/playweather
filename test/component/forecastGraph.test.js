import React from 'react';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import ForecastGraph from '../../src/components/forecastGraph';
import combinedReducer from '../../src/reducers';
import { forecast } from '../__testData__';

const props = {
  fetchData: () => { },
  text: 'scorpion',
  linkTo: 'scorpion king'
};


describe('ForecastGraph component without test', () => {
  it('renders correctly', () => {
    const store = createStore(combinedReducer(), { weatherReducer: { activeLocation: 'machetser, uk', forecast } });
    const actual = renderer.create(
      <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
        <Provider store={store}>
          <ForecastGraph {...props} />
        </Provider>
      </MemoryRouter>
    );
    expect(actual).toMatchSnapshot();
  });
});