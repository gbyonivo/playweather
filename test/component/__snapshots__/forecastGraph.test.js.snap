// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastGraph component without test renders correctly 1`] = `
<div
  className="forecastGraph"
>
  <div>
    <h2>
      Forecast Line Chart
    </h2>
    <div
      className="recharts-wrapper"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "default",
          "height": 300,
          "position": "relative",
          "width": 500,
        }
      }
    >
      <svg
        className="recharts-surface"
        height={300}
        version="1.1"
        viewBox="0 0 500 300"
        width={500}
      >
        <g
          className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
        >
          <line
            className="recharts-cartesian-axis-line"
            fill="none"
            height={30}
            stroke="#fff"
            width={430}
            x={65}
            x1={65}
            x2={495}
            y={265}
            y1={265}
            y2={265}
          />
          <g
            className="recharts-cartesian-axis-ticks"
          >
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                stroke="#fff"
                width={430}
                x={65}
                x1={65}
                x2={65}
                y={265}
                y1={271}
                y2={265}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#fff"
                height={30}
                stroke="none"
                textAnchor="middle"
                width={430}
                x={65}
                y={273}
              >
                <tspan
                  dy="0.71em"
                  x={65}
                >
                  05/11
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                stroke="#fff"
                width={430}
                x={65}
                x1={495}
                x2={495}
                y={265}
                y1={271}
                y2={265}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#fff"
                height={30}
                stroke="none"
                textAnchor="middle"
                width={430}
                x={495}
                y={273}
              >
                <tspan
                  dy="0.71em"
                  x={495}
                >
                  05/13
                </tspan>
              </text>
            </g>
          </g>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
        >
          <line
            className="recharts-cartesian-axis-line"
            fill="none"
            height={260}
            stroke="#fff"
            width={60}
            x={5}
            x1={65}
            x2={65}
            y={5}
            y1={5}
            y2={265}
          />
          <g
            className="recharts-cartesian-axis-ticks"
          >
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={260}
                stroke="#fff"
                width={60}
                x={5}
                x1={59}
                x2={65}
                y={5}
                y1={265}
                y2={265}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#fff"
                height={260}
                stroke="none"
                textAnchor="end"
                width={60}
                x={57}
                y={265}
              >
                <tspan
                  dy="0.355em"
                  x={57}
                >
                  0
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={260}
                stroke="#fff"
                width={60}
                x={5}
                x1={59}
                x2={65}
                y={5}
                y1={200}
                y2={200}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#fff"
                height={260}
                stroke="none"
                textAnchor="end"
                width={60}
                x={57}
                y={200}
              >
                <tspan
                  dy="0.355em"
                  x={57}
                >
                  3
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={260}
                stroke="#fff"
                width={60}
                x={5}
                x1={59}
                x2={65}
                y={5}
                y1={135}
                y2={135}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#fff"
                height={260}
                stroke="none"
                textAnchor="end"
                width={60}
                x={57}
                y={135}
              >
                <tspan
                  dy="0.355em"
                  x={57}
                >
                  6
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={260}
                stroke="#fff"
                width={60}
                x={5}
                x1={59}
                x2={65}
                y={5}
                y1={70}
                y2={70}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#fff"
                height={260}
                stroke="none"
                textAnchor="end"
                width={60}
                x={57}
                y={70}
              >
                <tspan
                  dy="0.355em"
                  x={57}
                >
                  9
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={260}
                stroke="#fff"
                width={60}
                x={5}
                x1={59}
                x2={65}
                y={5}
                y1={5}
                y2={5}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#fff"
                height={260}
                stroke="none"
                textAnchor="end"
                width={60}
                x={57}
                y={5}
              >
                <tspan
                  dy="0.355em"
                  x={57}
                >
                  12
                </tspan>
              </text>
            </g>
          </g>
        </g>
        <g
          className="recharts-cartesian-grid"
        >
          <g
            className="recharts-cartesian-grid-horizontal"
          >
            <line
              fill="none"
              height={260}
              stroke="#fff"
              strokeDasharray="1"
              width={430}
              x={65}
              x1={65}
              x2={495}
              y={5}
              y1={265}
              y2={265}
            />
            <line
              fill="none"
              height={260}
              stroke="#fff"
              strokeDasharray="1"
              width={430}
              x={65}
              x1={65}
              x2={495}
              y={5}
              y1={200}
              y2={200}
            />
            <line
              fill="none"
              height={260}
              stroke="#fff"
              strokeDasharray="1"
              width={430}
              x={65}
              x1={65}
              x2={495}
              y={5}
              y1={135}
              y2={135}
            />
            <line
              fill="none"
              height={260}
              stroke="#fff"
              strokeDasharray="1"
              width={430}
              x={65}
              x1={65}
              x2={495}
              y={5}
              y1={70}
              y2={70}
            />
            <line
              fill="none"
              height={260}
              stroke="#fff"
              strokeDasharray="1"
              width={430}
              x={65}
              x1={65}
              x2={495}
              y={5}
              y1={5}
              y2={5}
            />
          </g>
          <g
            className="recharts-cartesian-grid-vertical"
          >
            <line
              fill="none"
              height={260}
              stroke="#fff"
              strokeDasharray="1"
              width={430}
              x={65}
              x1={65}
              x2={65}
              y={5}
              y1={5}
              y2={265}
            />
            <line
              fill="none"
              height={260}
              stroke="#fff"
              strokeDasharray="1"
              width={430}
              x={65}
              x1={495}
              x2={495}
              y={5}
              y1={5}
              y2={265}
            />
          </g>
        </g>
        <g
          className="recharts-layer recharts-line"
        >
          <path
            className="recharts-curve recharts-line-curve"
            clipPath={null}
            d="M65,48.333333333333314L495,48.333333333333314"
            fill="none"
            height={260}
            stroke="#82ca9d"
            strokeDasharray="0px 0px"
            strokeWidth={1}
            width={430}
          />
        </g>
      </svg>
      <div
        className="recharts-legend-wrapper"
        style={
          Object {
            "bottom": 5,
            "height": "auto",
            "left": 5,
            "position": "absolute",
            "width": 490,
          }
        }
      >
        <ul
          className="recharts-default-legend"
          style={
            Object {
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
            }
          }
        >
          <li
            className="recharts-legend-item legend-item-0"
            style={
              Object {
                "display": "inline-block",
                "marginRight": 10,
              }
            }
          >
            <svg
              className="recharts-surface"
              height={14}
              style={
                Object {
                  "display": "inline-block",
                  "marginRight": 4,
                  "verticalAlign": "middle",
                }
              }
              version="1.1"
              viewBox="0 0 32 32"
              width={14}
            >
              <path
                className="recharts-legend-icon"
                d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                fill="none"
                stroke="#82ca9d"
                strokeWidth={4}
              />
            </svg>
            <span
              className="recharts-legend-item-text"
            >
              temp
            </span>
          </li>
        </ul>
      </div>
      <div
        className="recharts-tooltip-wrapper"
        style={
          Object {
            "MozTransform": "translate(undefinedpx, undefinedpx)",
            "OTransform": "translate(undefinedpx, undefinedpx)",
            "WebkitTransform": "translate(undefinedpx, undefinedpx)",
            "msTransform": "translate(undefinedpx, undefinedpx)",
            "pointerEvents": "none",
            "position": "absolute",
            "top": 0,
            "transform": "translate(undefinedpx, undefinedpx)",
            "visibility": "hidden",
          }
        }
      >
        <div
          className="recharts-default-tooltip"
          style={
            Object {
              "backgroundColor": "#fff",
              "border": "1px solid #ccc",
              "margin": 0,
              "padding": 10,
              "whiteSpace": "nowrap",
            }
          }
        >
          <p
            className="recharts-tooltip-label"
            style={
              Object {
                "margin": 0,
              }
            }
          >
            
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
